src/ffi/logging.rs

```diff
@@ -0,0 +1,232 @@
+use std::{
+    io::{self, ErrorKind},
+    str::from_utf8,
+    sync::{Arc, Mutex},
+};
+
+use tracing::{level_filters::LevelFilter, Subscriber};
+use tracing_subscriber::{
+    field::delimited::Delimited,
+    fmt::{self, format::debug_fn, FormatEvent, FormatFields, MakeWriter},
+    layer::SubscriberExt,
+    registry::LookupSpan,
+};
+
+use crate::{TelioLogLevel, TelioLoggerCb};
+
+pub fn build(log_level: crate::TelioLogLevel, logger: Box<dyn TelioLoggerCb>) -> impl Subscriber {
+    let field_fmt = Delimited::new(
+        " ",
+        debug_fn(|writer, field, value| {
+            if field.name() == "message" {
+                write!(writer, "{value:?}")
+            } else {
+                write!(writer, "{field}={value:?}")
+            }
+        }),
+    );
+
+    tracing_subscriber::registry()
+        .with(LevelFilter::from_level(log_level.into()))
+        .with(
+            fmt::layer()
+                .event_format(TelioEventFmt)
+                .fmt_fields(field_fmt)
+                .with_writer(FfiCallback::new(logger)),
+        )
+}
+
+struct TelioEventFmt;
+
+impl<S, N> FormatEvent<S, N> for TelioEventFmt
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+    N: for<'a> FormatFields<'a> + 'static,
+{
+    fn format_event(
+        &self,
+        ctx: &fmt::FmtContext<'_, S, N>,
+        mut writer: fmt::format::Writer<'_>,
+        event: &tracing::Event<'_>,
+    ) -> std::fmt::Result {
+        let meta = event.metadata();
+        write!(
+            writer,
+            "{:?}:{} ",
```

Comment:
Very minor nit: is `:?` needed here? It looks like it's a string that goes there and strings implement Display. Or am I missing something?

Reply:
This debug is not to break how it was displayed previously specificaly "some::path" instead of uquated "some::path"